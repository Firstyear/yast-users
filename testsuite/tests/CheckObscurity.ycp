{

    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);

    import "Users";

    define Test_pw (string username, string pw) ``{

        DUMP ("==========================================================");
        DUMP (sformat("username: %1, password: %2", username, pw));
        TEST(``(Users::CheckObscurity (username, pw)), [], 0);
    }

    Test_pw ("user1", "password");
    Test_pw ("user1", "uuser11!");
    Test_pw ("user1", "23534677");
    Test_pw ("user1", "us235467");
}
