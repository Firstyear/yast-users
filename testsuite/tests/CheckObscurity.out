Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Execute	.target.bash_output "locale -k charmap" $[]
Dump	==========================================================
Dump	username: user1, password: password
Return	You have used only lowercase letters for the password.
Dump	==========================================================
Dump	username: user1, password: uuser11!
Return	You have used the user name as a part of the password.
Dump	==========================================================
Dump	username: user1, password: 23534677
Return	You have used only digits for the password.
Dump	==========================================================
Dump	username: user1, password: us235467
Return	OK
